src/components/views/settings/tabs/user/VoiceUserSettingsTab.tsx

```diff
@@ -43,17 +44,24 @@ const getDefaultDevice = (devices: Array<Partial<MediaDeviceInfo>>) => {
 
 interface IState extends Record<MediaDeviceKindEnum, string> {
     mediaDevices: IMediaDevices;
+    audioAutoGainControl: boolean;
+    audioEchoCancellation: boolean;
+    audioNoiseSuppression: boolean;
```

Comment:
I wonder if we could use `extends` here somehow

Reply:
I'm not really familiar with TypeScript, but it seems to me that I'd violate the Liskov substitution principle with that, as audio-related settings do not seem to be in an "IS-A" relation with `VoiceUserSettingsTab`'s state.
I'd apply "composition over inheritance" in such cases.

Reply:
Ok, that makes sense but I think we should also change `MediaDeviceKindEnum` then